src/Files.App/UserControls/Widgets/HomePageWidget.cs

```diff
@@ -13,51 +13,81 @@ namespace Files.App.UserControls.Widgets
 {
 	public abstract class HomePageWidget : UserControl
 	{
+		// Dependency injections
+
 		public IUserSettingsService UserSettingsService { get; } = Ioc.Default.GetRequiredService<IUserSettingsService>();
 		public IQuickAccessService QuickAccessService { get; } = Ioc.Default.GetRequiredService<IQuickAccessService>();
 		public IStorageService StorageService { get; } = Ioc.Default.GetRequiredService<IStorageService>();
 
-		public ICommand RemoveRecentItemCommand;
-		public ICommand ClearAllItemsCommand;
-		public ICommand OpenFileLocationCommand;
-		public ICommand OpenInNewTabCommand;
-		public ICommand OpenInNewWindowCommand;
-		public ICommand OpenPropertiesCommand;
-		public ICommand PinToFavoritesCommand;
-		public ICommand UnpinFromFavoritesCommand;
+		// Fields
+
+		protected string? FlyoutItemPath;
+
+		// Commands
+
+		public ICommand? RemoveRecentItemCommand { get; protected set; }
+		public ICommand? ClearAllItemsCommand { get; protected set; }
+		public ICommand? OpenFileLocationCommand { get; protected set; }
+		public ICommand? OpenInNewTabCommand { get; protected set; }
+		public ICommand? OpenInNewWindowCommand { get; protected set; }
+		public ICommand? OpenPropertiesCommand { get; protected set; }
+		public ICommand? PinToFavoritesCommand { get; protected set; }
+		public ICommand? UnpinFromFavoritesCommand { get; protected set; }
+
+		// Events
 
-		protected CommandBarFlyout ItemContextMenuFlyout;
-		protected string FlyouItemPath;
+		public static event EventHandler<WidgetsRightClickedItemChangedEventArgs>? RightClickedItemChanged;
+
+		// Abstract methods
 
 		public abstract List<ContextMenuFlyoutItemViewModel> GetItemMenuItems(WidgetCardItem item, bool isPinned, bool isFolder = false);
 
+		// Event methods
+
 		public void Button_RightTapped(object sender, RightTappedRoutedEventArgs e)
 		{
-			var itemContextMenuFlyout = new CommandBarFlyout { Placement = FlyoutPlacementMode.Full };
-			itemContextMenuFlyout.Opening += (sender, e) => App.LastOpenedFlyout = sender as CommandBarFlyout;
-			if (sender is not Button widgetCardItem || widgetCardItem.DataContext is not WidgetCardItem item)
+			// Ensure values are not null
+			if (sender is not Button widgetCardItem ||
+				widgetCardItem.DataContext is not WidgetCardItem item)
 				return;
 
+			// Create a new Flyout
+			var itemContextMenuFlyout = new CommandBarFlyout()
+			{
+				Placement = FlyoutPlacementMode.Full
+			};
+
+			// Hook events
+			itemContextMenuFlyout.Opening += (sender, e) => App.LastOpenedFlyout = sender as CommandBarFlyout;
```

Comment:
Does this get garbage-collected?

Reply:
I dont know...

Reply:
If it's not needed I'd advise to remove it. It's definitely not getting GC'd

Reply:
I’m gonna try to find who added this line and remove if it seems to be ok.

Reply:
We added it to dismiss open context menus when closing the app, otherwise we found that it could cause the app to hang.

Reply:
Fyi @hishitetsu

Reply:
It's a workaround for #10110. This workaround is still necessary to avoid crashes.